src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/actions/select/RunSelectedProjectsAction.java

```diff
@@ -0,0 +1,60 @@
+package com.facebook.buck.intellij.ideabuck.actions.select;
+
+import com.facebook.buck.intellij.ideabuck.build.BuckBuildCommandHandler;
+import com.facebook.buck.intellij.ideabuck.build.BuckBuildManager;
+import com.facebook.buck.intellij.ideabuck.build.BuckCommand;
+import com.facebook.buck.intellij.ideabuck.config.BuckModule;
+import com.intellij.ide.IdeView;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.actionSystem.LangDataKeys;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiDirectory;
+
+/**
+ * Action to generate projects from the project tool view.
+ */
+public class RunSelectedProjectsAction extends AnAction {
+
+  @Override
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    final Project project = anActionEvent.getProject();
+    if (project == null) {
+      return;
+    }
+
+    BuckModule buckModule = project.getComponent(BuckModule.class);
+    DataContext dataContext = anActionEvent.getDataContext();
+    final IdeView view = LangDataKeys.IDE_VIEW.getData(dataContext);
+    if (view == null) {
+      return;
+    }
+
+    final PsiDirectory directory = view.getOrChooseDirectory();
+    if (directory == null) {
```

Comment:
if its done on a file, we can use buck query to see which targets own the file as well